backend/src/services/player.service.ts

```diff
@@ -10,6 +10,16 @@ export const createPlayer = async (
   token: Player["token"];
 }> => {
   const game = await getGame(gameCode);
+  // TODO - add the const for MaxPlayers defined when the discard shuffle PR is merged
+  const gameMax =
+    game.settings !== undefined && game.settings.maxPlayers !== undefined
+      ? game.settings.maxPlayers
+      : 40;
+  if (game.state === GameState.finished) {
+    throw new ServiceError(ErrorType.gameError, "Game is finished");
+  } else if (game.players.length >= gameMax) {
+    throw new ServiceError(ErrorType.gameError, "Too many players in the game");
+  }
```

Comment:
Since we're using Modern JavaScript you can use [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining)!
 
Try: `const gameMax = game?.settings?.maxPlayers || 40`!

Reply:
Done!